addons/web/static/src/core/macro.js

```diff
@@ -171,26 +235,26 @@ export class MacroEngine {
         if (this.timeout) {
             browser.clearTimeout(this.timeout);
         }
-        this.timeout = browser.setTimeout(this.advanceMacros.bind(this), this.interval);
+        this.timeout = browser.setTimeout(
+            () => mutex.exec(this.advanceMacros.bind(this)),
```

Comment:
not sure why we need the mutex? seems to me that it is not necessary.

Reply:
The mutex is needed to respect the step_delay. Without it, even if the user puts a step delay in the tour (say 1sec), the tour will still appear to be running as fast as possible.